Collaboration2/Collaboration2/Scenes/Weather Page/Picker,DetailsAndDailyViews/DailyWeatherView.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  DailyWeatherView.swift
+//  Collaboration2
+//
+//  Created by Sandro Gelashvili on 12.06.24.
+//
+
+import SwiftUI
+
+struct DailyWeatherView: View {
+    @ObservedObject var viewModel: WeatherPageViewModel
+    
+    var body: some View {
+        ZStack {
+            VStack() {
+                LocationPicker()
+                    .offset(x: 140)
+                
+                DetailsInfoForCurrentWeather(viewModel: viewModel)
+                
+            VStack(alignment: .leading) {
+                ForEach(DaysOfWeek.allCases, id: \.self) { day in
+                    WeeklyWeatherCell(imageForWeatherIcon: "image",
+                                      maxTemp: 20,
+                                      minTemp: 15,
+                                      day: day)
+                }
+            }
+            .background(.ultraThinMaterial,
+                        in: RoundedRectangle(cornerRadius: 20,
+                                             style: .continuous))
+            .padding()
+        }
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(LinearGradient(colors: [Color(#colorLiteral(red: 0.1897571981, green: 0.6920755506, blue: 0.8686286211, alpha: 1)), Color(#colorLiteral(red: 0.55633533, green: 0.6781123877, blue: 0.8838557601, alpha: 1))],
+                                   startPoint: .top,
+                                   endPoint: .bottom))
+        .edgesIgnoringSafeArea(.all)
+    }
+    
+    private struct WeeklyWeatherCell: View {
+        var imageForWeatherIcon: String
+        var maxTemp: Int
+        var minTemp: Int
+        let day: DaysOfWeek
+        
+        var body: some View {
+            HStack () {
+                Text(day.rawValue)
+                    .frame(width: 97,alignment: .leading)
+                
+                Spacer()
+                
+                AsyncImageForIcon(imageForWeatherIcon: imageForWeatherIcon)
+                
+                Spacer()
+                
+                Text("\(maxTemp)°C")
+                Text("\(minTemp)°C")
+                    .foregroundStyle(Color(.systemGray5))
+            }
+            .font(.system(size: 18, weight: .medium))
+            .foregroundStyle(.white)
+            .padding()
+        }
+    }
+    
+    private struct AsyncImageForIcon: View {
```

Comment:
consider moving this struct outside of WeeklyWeatherCell. it might make WeeklyWeatherCell less ovrewhelming to look at and thinner